download/tests/read-proxy-env.rs

```diff
@@ -0,0 +1,50 @@
+#![cfg(feature = "reqwest-backend")]
+
+use std::env::{remove_var, set_var};
+use std::sync::Mutex;
+
+use env_proxy::for_url;
+use lazy_static::lazy_static;
+use url::Url;
+
+lazy_static! {
+    static ref LOCK: Mutex<()> = Mutex::new(());
+}
+
+fn scrub_env() {
+    remove_var("http_proxy");
+    remove_var("https_proxy");
+    remove_var("HTTPS_PROXY");
+    remove_var("ftp_proxy");
+    remove_var("FTP_PROXY");
+    remove_var("all_proxy");
+    remove_var("ALL_PROXY");
+    remove_var("no_proxy");
+    remove_var("NO_PROXY");
+}
+
+// Tests for:
+// 1. Correctly retrieving the proxy (host, port) tuple from $https_proxy or $HTTPS_PROXY
+// 2. Prioritizing $https_proxy over $HTTPS_PROXY (except on Windows, where env vars
```

Comment:
I don't think point 2 is interesting: this is very much in the space of 'checking env_proxy internals' :  the point of this test is to make sure that urls hooks <-> env_proxy are setup properly in case of future dependency skew issues, not to test the definition of env-proxy.

Reply:
Ack, just checking `https_proxy` is probably enough.